doc/source/train/getting-started-xgboost.rst

```diff
@@ -0,0 +1,429 @@
+                # should configure the run's persistent storage that is accessible
+                # across all worker nodes.
+                # run_config=ray.train.RunConfig(storage_path="s3://..."),
+            )
+            result = trainer.fit()
+
+            # 7. Load the trained model
+            import os
+            with result.checkpoint.as_directory() as checkpoint_dir:
+                model_path = os.path.join(checkpoint_dir, RayTrainReportCallback.CHECKPOINT_NAME)
+                model = xgboost.Booster()
+                model.load_model(model_path)
+
+
+Set up a training function
+--------------------------
+
+First, update your training code to support distributed training.
+Begin by wrapping your code in a :ref:`training function <train-overview-training-function>`:
+
+.. testcode::
+    :skipif: True
+
+    def train_func():
+        # Your model training code here.
+        ...
+
+Each distributed training worker executes this function.
+
+You can also specify the input argument for `train_func` as a dictionary via the Trainer's `train_loop_config`. For example:
+
+.. testcode:: python
+    :skipif: True
+
+    def train_func(config):
+        label_column = config["label_column"]
+        num_boost_round = config["num_boost_round"]
+        ...
+
+    config = {"label_column": "y", "num_boost_round": 10}
+    trainer = ray.train.xgboost.XGBoostTrainer(train_func, train_loop_config=config, ...)
+
+.. warning::
+
+    Avoid passing large data objects through `train_loop_config` to reduce the
+    serialization and deserialization overhead. Instead, it's preferred to
+    initialize large objects (e.g. datasets, models) directly in `train_func`.
+
+    .. code-block:: diff
+
+         def load_dataset():
+             # Return a large in-memory dataset
+             ...
+
+         def load_model():
+             # Return a large in-memory model instance
+             ...
+
+        -config = {"data": load_dataset(), "model": load_model()}
+
+         def train_func(config):
+        -    data = config["data"]
+        -    model = config["model"]
+
+        +    data = load_dataset()
+        +    model = load_model()
+             ...
+
+         trainer = ray.train.xgboost.XGBoostTrainer(train_func, train_loop_config=config, ...)
+
+Ray Train automatically sets up the Rabit communicator for XGBoost, which handles the distributed communication between workers.
+
+Report metrics and save checkpoints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To persist your checkpoints and monitor training progress, add a
+:class:`ray.train.xgboost.RayTrainReportCallback` utility callback to your Trainer:
+
+
+.. code-block:: diff
+
+     import xgboost
+     from ray.train.xgboost import RayTrainReportCallback
+
+     def train_func():
+        ...
+        bst = xgboost.train(
+            ...,
+            callbacks=[
+                RayTrainReportCallback(
+                    metrics={"loss": "eval-logloss"}, frequency=1
```

Comment:
the dict renaming map thing is a bit advanced/confusing. let's just keep the metric names that are returned by xgboost.

```suggestion
                RayTrainReportCallback(
                    metrics=["eval-logloss"], frequency=1
```